### PR TITLE
Script to rename files so the case matches the module name

### PR DESCRIPTION
**Pull request by ahocevar**
This pull request modifies the `npm run transform` script so the module file names match the case of the default export.

### CODE REVIEW

**Comment by gberaudo** [comment]
@ahocevar, great to see this change.
Do you need the script part? I would expect the names to be correct after your change in the transform.

**Comment by marcjansen** [comment]
Will this mean that users of the ol package have to change their imports? If so we should tell people this, right?

**Comment by ahocevar** [comment]
@marcjansen I need to change this pull request. It was meant to accompany #7181, and not make changes to the ol package for now. For changes people need to make once what we've outlined in #7181 is in place, we are going to create a roadmap page with instructions on the wiki.

**Comment by ahocevar** [comment]
Pull request updated.

File: package.json [hunk 1]
```diff
@@ -19,9 +19,10 @@
     "debug-server": "node tasks/serve-lib.js",
     "karma": "node tasks/test.js start test/karma.config.js",
     "transform-src": "jscodeshift --transform transforms/module.js src",
+    "changecase-src": "node tasks/filename-case-from-module.js",
     "transform-examples": "jscodeshift --transform transforms/module.js examples",
     "transform-test": "jscodeshift --transform transforms/module.js test",
-    "transform": "npm run transform-src && npm run transform-examples && npm run transform-test && npm run lint -- --fix"
+    "transform": "npm run transform-src && npm run changecase-src && npm run transform-examples && npm run transform-test && npm run lint -- --fix"
```
Comment (tschaub) [inline package.json]: I think we should be changing the case of file names *before* running the module transform on them.  In the end, we want `src/ol/Map.js` to look like:
```js
/**
 * @module ol/Map
 */
import _ol_View_ from './View.js';
// etc.
```

Comment (ahocevar) [reply package.json]: Good point. Changing.

Comment (ahocevar) [reply package.json]: Done, commit added.

**Comment by ahocevar** [comment]
Strange Travis issue when building api docs. I'm sure it's unrelated to this pull request, but am unable to investigate this now.

**Comment by tschaub** [comment]
See https://github.com/jsdoc3/jsdoc/issues/1438 for the build failure.

**Comment by ahocevar** [comment]
Thanks @tschaub for investigating the Travis issue. I've set the node version to 8.4 in `.travis.yml` for now.

**Comment by tschaub** [comment]
> Will this mean that users of the ol package have to change their imports? If so we should tell people this, right?

@marcjansen - you were right.  We should have left out the changes to `transforms/module.js` until 5.0.  I published `ol@4.4.0` with this change before realizing that it is breaking (on case sensitive filesystems).  I put up #7313 to fix and will get out a patch release when that is in.

**Comment by tschaub** [comment]
In the meantime, `ol@4.4.1-beta.1` is out (I can't wait until the whole release process is as easy as publishing the `ol` package is now).

**Pull request merged**